be/src/exec/pipeline/hashjoin/hash_join_build_operator.cpp

```diff
@@ -37,19 +35,20 @@ Status HashJoinBuildOperator::push_chunk(RuntimeState* state, const ChunkPtr& ch
 Status HashJoinBuildOperator::prepare(RuntimeState* state) {
     RETURN_IF_ERROR(Operator::prepare(state));
 
+    _partial_rf_merger->incr_builder();
+
+    // For prober.
+    // HashJoinProbeOperator may be instantiated lazily, so join_builder is ref here
+    // and unref when all the probers are finished in join_builder->decr_prober.
+    _join_builder->ref();
```

Comment:
Yes, we cannot ref `join_builder` in `HashJoinProbeOperator`, because `HashJoinProbeOperator` may be created later. If we don't ref `join_builder` twice here, `join_builder` will be released when `HashJoinBuildOperator` is finished and `HashJoinProbeOperator` hasn't been created.